src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -32,7 +32,7 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
             builder.UseMiddleware<FunctionInvocationMiddleware>();
             builder.UseMiddleware<HostWarmupMiddleware>();
 
-            builder.UseWhen(context => !context.Request.Path.StartsWithSegments("/admin"), config =>
+            builder.UseWhen(context => !context.Request.Path.StartsWithSegments("/admin") && !context.Request.Path.Equals("/"), config =>
```

Comment:
Not sure why were are running the host availability check on the home page root? We shouldn't require the host to be up and running when hitting the root. Also, it is causing my offline short circuit in HostAvailabilityCheckMiddleware to blow up/not work.

Reply:
I don't think this was intentional. Although, the root pings used to rely on the response returning to assume the  host was up and running, so it is possible this was by design.

Reply:
I was thinking it might be related to proxies. We'll need to review all this.